### PR TITLE
(#889) Tutorial: Creating a script package

### LINKED ISSUE #889: Tutorial - Create a Script Package
## What New Or Updated Would You Like To See?

We need a prescriptive walk-through for creating Chocolatey Script Package

## Why Is It Needed?

We don't have any easy to follow instructions for creating this package type.

### PR DESCRIPTION
**Pull request by steviecoaster**
## Description Of Changes

Adds a tutorial for creating a script-type Chocolatey Package

## Motivation and Context

Making documentation for creating different types of packaging approachable is important to our end users.

## Testing

* [x] I have previewed these changes using the [Docker Container](https://github.com/chocolatey/docs/tree/master/.devcontainer) or another method before submitting this pull request.

## Change Types Made

* [ ] Minor documentation fix (typos etc.).
* [ ] Major documentation change (refactoring, reformatting or adding documentation to existing page).
* [x] New documentation page added.
* [ ] The change I have made should have a video added, and I have raised an issue for this.
    * Issue #

## Change Checklist

* [ ] Requires a change to menu structure (top or left hand side)/
* [ ] Menu structure has been updated

## Related Issue

Fixes #889

### CODE REVIEW

File: input/en-us/guides/create/create-script-package.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+This powerful package type allows you to version system configuration, making reporting and compliance much simpler!
+
+If you have not yet read [Preparing Your Environment for Package Creation](#placeholder), please do so now and ensure
+you are ready to go before continuing.
+
+### Creating Your Package
+
+1. Open the tutorials folder in VSCode
+2. Press `Ctrl+Shift+P` or select View > Command Palette
+3. Type `Chocolatey:` and select `Create new Chocolatey package` from the list of available commands
```
Comment (gep13) [inline input/en-us/guides/create/create-script-package.md:18]: Am I right in saying that you are going to mention the installation of the Chocolatey VSCode extension in the `Preparing Your Environment for Package Creation` link?

File: input/en-us/guides/create/create-script-package.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  <files>
+    <!-- this section controls what actually gets packaged into the Chocolatey package -->
+    <file src="tools\**" target="tools" />
+  </files>
+</package>
+```
+
+### Compile Your Package
+
+The `choco pack` is used to compile your Chocolatey package into a nupkg file.
```
Comment (gep13) [inline input/en-us/guides/create/create-script-package.md:86]: Not sure which way round is better, but I think something needs to change here..

```suggestion
The `choco pack` command is used to compile your Chocolatey package into a nupkg file.
```

```suggestion
`choco pack` is used to compile your Chocolatey package into a nupkg file.
```

**Review by gep13 — CHANGES_REQUESTED** [review]

File: input/en-us/guides/create/create-script-package.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+This tutorial will focus on creating a Chocolatey script package. This type of package is one which contains one or more PowerShell scripts that are executed by Chocolatey on the target system.
+This powerful package type allows you to version system configuration, making reporting and compliance much simpler!
+
+If you have not yet read [Preparing Your Environment for Package Creation](#placeholder), please do so now and ensure
+you are ready to go before continuing.
+
+### Creating Your Package
+
+1. Open the tutorials folder in VSCode
+2. Press `Ctrl+Shift+P` or select View > Command Palette
```
Comment (pauby) [inline input/en-us/guides/create/create-script-package.md:17]:
```suggestion
2. Press `Ctrl+Shift+P` or select **View** > **Command Palette**.
```

File: input/en-us/guides/create/create-script-package.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+This tutorial will focus on creating a Chocolatey script package. This type of package is one which contains one or more PowerShell scripts that are executed by Chocolatey on the target system.
+This powerful package type allows you to version system configuration, making reporting and compliance much simpler!
+
+If you have not yet read [Preparing Your Environment for Package Creation](#placeholder), please do so now and ensure
+you are ready to go before continuing.
+
+### Creating Your Package
+
+1. Open the tutorials folder in VSCode
```
Comment (pauby) [inline input/en-us/guides/create/create-script-package.md:16]:
```suggestion
1. Open the tutorials folder in VSCode.
```

File: input/en-us/guides/create/create-script-package.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+2. Press `Ctrl+Shift+P` or select View > Command Palette
+3. Type `Chocolatey:` and select `Create new Chocolatey package` from the list of available commands
+4. Give your package a name, e.g. `script-package`
+5. Select `Default Template` when prompted
```
Comment (pauby) [inline input/en-us/guides/create/create-script-package.md:20]:
```suggestion
4. Give your package a name, e.g. `script-package`.
5. Select `Default Template` when prompted.
```

File: input/en-us/guides/create/create-script-package.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+Write-Host "Executing script: $script"
+& $script
+```
+
+Save and close the file.
+
+### Cleaning Up The Packaging
+
+Our simple example here doesn't require anything special for an upgrade or uninstall scenario. In the VSCode explorer window find both the `chocolateyBeforeModify.ps1` and `chocolateyUninstall.ps1` files and remove them.
```
Comment (pauby) [inline input/en-us/guides/create/create-script-package.md:46]:
```suggestion
Our simple example here doesn't require anything special for an upgrade or uninstall scenario. In the VSCode Explorer window find both the `chocolateyBeforeModify.ps1` and `chocolateyUninstall.ps1` files and remove them.
```

File: input/en-us/guides/create/create-script-package.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+### Creating Your Nuspec File
```
Comment (pauby) [inline input/en-us/guides/create/create-script-package.md:48]:
```suggestion
### Creating The Package Metadata
```

Comment (pauby) [reply input/en-us/guides/create/create-script-package.md:48]: I feel the terms we use, nupkg and nuspec, don't heop make things easier for peopel to understand. They sound very complex, acronyms and technical. So this is my suggestion to help with that.

File: input/en-us/guides/create/create-script-package.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+The nuspec file is the metadata for your package. It provides information to Chocolatey such as:
```
Comment (pauby) [inline input/en-us/guides/create/create-script-package.md:50]:
```suggestion
The package metadata is stored in a file with a `.nuspec`extension. It provides information to Chocolatey such as:
```

Comment (pauby) [reply input/en-us/guides/create/create-script-package.md:50]: This is just a follow-on from the above.

File: input/en-us/guides/create/create-script-package.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+- Package Id
+- Package Version
+- Author
+- Synopsis
+- Dependencies
+
+For packages being published to the [Chocolatey Community Repository](https://community.chocolatey.org/packages) additional information is required.
+You can find information on nuspec requirements in our [Package Validator Rules](http://localhost:5080/en-us/community-repository/moderation/package-validator/rules/#requirements) documentation.
```
Comment (pauby) [inline input/en-us/guides/create/create-script-package.md:59]:
```suggestion
You can find information on package metadata requirements in our [Package Validator Rules](http://localhost:5080/en-us/community-repository/moderation/package-validator/rules/#requirements documentation.
```

Comment (pauby) [reply input/en-us/guides/create/create-script-package.md:59]: Following on from above. Also note the URL above is wrong (localhost?). I haven't fixed it as you'll likely need to use the xref for it.

File: input/en-us/guides/create/create-script-package.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+
+In the explorer window find and open the `embedded-zip.nuspec` file.
+
+Replace the contents of the nuspec file with the following:
```
Comment (pauby) [inline input/en-us/guides/create/create-script-package.md:63]:
```suggestion
In the VS Code Explorer window, find and open the `embedded-zip.nuspec` file.

Replace the contents of the file with the following:
```

File: input/en-us/guides/create/create-script-package.md (hunk 2 above)
Comment (pauby) [reply input/en-us/guides/create/create-script-package.md:86]:
```suggestion
The `choco pack` is used to compile your package files into a usable Chocolatey package.
```

File: input/en-us/guides/create/create-script-package.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+    <!-- this section controls what actually gets packaged into the Chocolatey package -->
+    <file src="tools\**" target="tools" />
+  </files>
+</package>
+```
+
+### Compile Your Package
+
+The `choco pack` is used to compile your Chocolatey package into a nupkg file.
+1. In VSCode press `ctrl+shift+p` or `View > Command Pallete`
```
Comment (pauby) [inline input/en-us/guides/create/create-script-package.md:87]:
```suggestion
1. In VSCode press `Ctrl+Shift+P` or **View** > **Command Pallete**
```

File: input/en-us/guides/create/create-script-package.md [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+2. Type `Chocolatey:` and click `Chocolatey: Pack Chocolatey package(s)`
+3. Select `script-package.nuspec` from the list
+4. In the `additional arguments` dialog enter `--output-directory='C:\tutorials'` and press `Enter`
```
Comment (pauby) [inline input/en-us/guides/create/create-script-package.md:90]:
```suggestion
4. In the **Additional arguments** dialog enter `--output-directory='C:\tutorials'` and press `Enter`.
```

**Review by pauby — CHANGES_REQUESTED** [review]

File: input/en-us/guides/create/create-script-package.md (hunk 1 above)
Comment (steviecoaster) [reply input/en-us/guides/create/create-script-package.md:18]: Yes, Preparing your environment covers the installation of VSCode and all the needed extensions.

**Comment by pauby** [comment]
@steviecoaster I've made a couple of tweaks to this including renaming the script file (to make it clearer what it's for). Can you check and approve? I can merge.

**Review by pauby — APPROVED** [review]

**Pull request merged**